proctor-common/src/main/java/com/indeed/proctor/common/RuleEvaluator.java

```diff
@@ -169,6 +170,26 @@ static void checkRuleIsBooleanType(
         // To support users writing rules, be more strict here in requiring the type of the
         // value to be expected before coercion
         Class<?> type = ve.getType(elContext);
+
+        // if object is map check what the base value type is
+        if (type == Object.class) {
```

Comment:
I'm not familiar with el; when would this be false?

Reply:
The Type can be any class, it is the type of the variables in the rules. From what I have seen it will return object for Maps. The get Type will try to figure out what the return of the rule is without plugging in the context variables. So `X == Y` will be a boolean but just `map.key` will be an object

Reply:
I see; so if type is a boolean, it will NOT match `Boolean.class` (it will match `Boolean.class`). Is there any case where `Object.class` is returned but it isn't a map?

Reply:
I am not 100%. My logic should handle only performing this step for maps by checking what is contained in the Node within the value expression. 

Other things that could be objects would be other collection types or custom classes. I think we can handle other cases in future MRs if they ever end up getting used in rules. I think currently we do not need to worry about the other cases.

Reply:
got it! LGTM!